pkg/blockbuilder/tsdb.go

```diff
@@ -256,6 +259,15 @@ func (b *TSDBBuilder) newTSDB(tenant tsdbTenant) (*userTSDB, error) {
 	userID := tenant.tenantID
 	userLogger := util_log.WithUserID(userID, b.logger)
 
+	udb := &userTSDB{
+		userID: userID,
+	}
+
+	userLimit := b.limits.MaxGlobalSeriesPerUser(userID)
+	if userLimit <= b.applyGlobalSeriesLimitUnder {
+		udb.maxGlobalSeries = userLimit
```

Comment:
This new code is kinda mysterious on casual read. I think it needs a brief handrail comment about why we're selectively applying limits based on some threshold.